tree-sitter/src/TreeSitter/Unmarshal/Examples.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE DeriveAnyClass, DeriveGeneric, DuplicateRecordFields, TypeOperators #-}
+module TreeSitter.Unmarshal.Examples () where
+
+import qualified Data.Text as Text
+import GHC.Generics ((:+:), Generic)
+import TreeSitter.Unmarshal
+
+-- | An example of a sum-of-products datatype.
+data Expr a
+  = IfExpr (If a)
+  | BlockExpr (Block a)
+  | LitExpr (Lit a)
+  | BinExpr (Bin a)
```

Comment:
(When we implement #210, we’ll want to come back and replace this with a type synonym instead.)